projects/plugins/jetpack/extensions/blocks/map/mapkit-utils/index.js

```diff
@@ -54,9 +56,83 @@ function createCalloutElementCallback( currentDoc, callback ) {
 	};
 }
 
+function waitUntilMapkitIsInitialized( currentWindow ) {
+	return new Promise( resolve => {
+		const check = () => {
+			if ( currentWindow.mapkitIsInitializing === false ) {
+				resolve();
+			} else {
+				currentWindow.requestAnimationFrame( check );
+			}
+		};
+		check();
+	} );
+}
+
+function loadMapkitLibrary( currentDoc, currentWindow ) {
+	return new Promise( resolve => {
+		if ( currentWindow.mapkitScriptIsLoading ) {
+			waitForObject( currentWindow, 'mapkit' ).then( mapkitObj => {
+				resolve( mapkitObj );
+			} );
+		} else {
+			currentWindow.mapkitScriptIsLoading = true;
+
+			const element = currentDoc.createElement( 'script' );
+			element.addEventListener(
+				'load',
+				async () => {
+					const mapkitObj = await waitForObject( currentWindow, 'mapkit' );
+					currentWindow.mapkitScriptIsLoading = false;
+
+					resolve( mapkitObj );
+				},
+				{ once: true }
+			);
+			element.src = 'https://cdn.apple-mapkit.com/mk/5.x.x/mapkit.js';
+			element.crossOrigin = 'anonymous';
+			currentDoc.head.appendChild( element );
+		}
+	} );
+}
+
+function fetchMapkitKey( mapkitObj, blogId, currentWindow ) {
```

Comment:
I think it would be more appropriate to call the function `mapkitInitialize` (or `initiliazeMapkit`) after looking at the function's body. However I understand this can be subjective therefore I will leave it up to you to decide what is more suitable. 🙂

Reply:
I added this to the refactor/cleanup ticket: https://github.com/Automattic/wp-calypso/issues/74103 

It will be easier to merge if this happens later :)